x/incentives/client/cli/query.go

```diff
@@ -384,31 +386,80 @@ $ %s query incentives rewards-estimation
 				return err
 			}
 
-			lockIdStrs := strings.Split(lockIdsCombined, ",")
-			lockIds := []uint64{}
-			for _, lockIdStr := range lockIdStrs {
-				lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+			var res *lockuptypes.AccountLockedLongerDurationResponse
+			if owner != "" {
+				queryClientLockup := lockuptypes.NewQueryClient(clientCtx)
+
+				res, err = queryClientLockup.AccountLockedLongerDuration(cmd.Context(), &lockuptypes.AccountLockedLongerDurationRequest{Owner: owner, Duration: time.Millisecond})
 				if err != nil {
 					return err
 				}
-				lockIds = append(lockIds, lockId)
+			} else {
+				owner = ""
+			}
+
+			ownerLocks := []uint64{}
+
+			if res != nil {
+				for _, lockId := range res.Locks {
+					ownerLocks = append(ownerLocks, lockId.ID)
+				}
+			}
+
+			lockIdStrs := strings.Split(lockIdsCombined, ",")
+			lockIds := []uint64{}
+			// if user doesn't provide any lockIDs or a lock owner, we don't have enough information to proceed
+			if lockIdsCombined == "" && owner == "" {
+				err = fmt.Errorf("if owner flag is not set, lock IDs must be provided")
```

Comment:
Current wording implies the presence of lock ids is conditional on the lack of owner but not vice versa
```suggestion
				err = fmt.Errorf("either one of owner flag or lock IDs must be provided")
```

Reply:
Correct! If an owner flag is set, lock ids are optional. If an owner flag is not set, lock ids are required. But both owner and lock ids can also be provided

Reply:
So the suggestion applies, right?

```
if owner flag is not set, lock IDs must be provided
```
means `if !owner -> lock ids` but we also want the text to convey `if !lock ids -> owner`

Reply:
Yeah you are right, I had to read it like 30 times to make sure but you are right haha

Reply:
It's not allowing me to commit the suggestion since its "out of date" and I still want you to get credit for the suggestion, can you resubmit? If you don't care I will happily commit it myself.